sdks/unreal/Agones/Source/Agones/Private/AgonesComponent.cpp

```diff
@@ -83,6 +99,36 @@ void UAgonesComponent::Connect()
 	ConnectDel.BindUObject(this, &UAgonesComponent::GameServer, SuccessDel, FAgonesErrorDelegate());
 	GetWorld()->GetTimerManager().ClearTimer(ConnectDelTimerHandle);
 	GetWorld()->GetTimerManager().SetTimer(ConnectDelTimerHandle, ConnectDel, 5.f, true);
+

```

Comment:
would move this out of the connect as well and instead have somethin like the `WatchGameServer` func take in a delegate that will start off the watching of the state changes.

The `go` SDK does this with callbacks https://github.com/googleforgames/agones/blob/main/sdks/go/sdk.go#L131

Therefore the call to starting a watch is explicit you could also then get rid of `bWatchGameServer` and kick off all the things (connection keep alives etc.) from `WatchGameServer` rather than other parts of the code.